web/src/graph-viewer.ts

```diff
@@ -94,61 +93,85 @@ export class GraphViewer extends LitElement {
   }
 
   render() {
+    if (this.queryResult == null) return null;
+
+    if (this._errorMsg != null)
+      return html`<div id="error">${this._errorMsg}</div>`;
+
     return html`
       <sl-tab-group>
-        <sl-tab slot="nav" panel="${TABS.graph}">Graph</sl-tab>
-        <sl-tab slot="nav" panel="${TABS.raw}">Text</sl-tab>
+        <sl-tab
+          slot="nav"
+          panel="${TABS.graph}"
+          ?active=${this._currentTab === TABS.graph}
+          >Graph</sl-tab
+        >
+        <sl-tab
+          slot="nav"
+          panel="${TABS.raw}"
+          ?active=${this._currentTab === TABS.raw}
+          >Text</sl-tab
+        >
       </sl-tab-group>
 
       ${choose<TabType, TemplateResult>(this._currentTab, [
         [
           TABS.graph,
           () => html`
             <div id="cy-container">
-              ${this._error
-                ? html`<div id="error">Error fetching graph data :(</div>`
-                : html`<div id="cy"></div>`}
+              <div id="cy"></div>
             </div>
           `,
         ],
         [
           TABS.raw,
-          () => html` <div id="raw-data">${this.#graphData.raw}</div> `,
+          () => html`<div id="raw-data">${this.queryResult.data.raw}</div>`,
         ],
       ])}
     `;
   }
 
   async updated(changedProperties: Map<string, any>) {
     if (
-      this._error === false &&
+      this._errorMsg != null &&
```

Comment:
Did you mean this?
```suggestion
      this._errorMsg === null &&
```
It looks weird to me to go from "if there is no error" to "if there is an error message"

Reply:
The comparison is ok, but I also think that is confusing. While adding pagination, I noticed similar issues with the `nix-search` component, the logic is hard to understand.

I'll try to fix both in a different PR, for now I added a comment to the `_errorMsg` definition to make it a bit less confusing.